configure.ac

```diff
@@ -153,6 +154,35 @@ AS_IF([test "x$with_nvme" != "xno"],
       [AC_DEFINE([WITH_BD_NVME], [], [Define if nvme is supported]) AC_SUBST([WITH_NVME], [1])],
       [])
 
+AC_SUBST([WITH_SMART], [0])
+AC_ARG_WITH([libatasmart],
+    AS_HELP_STRING([--with-libatasmart], [support ATA S.M.A.R.T. via libatasmart @<:@default=yes@:>@]),
+    [],
+    [with_libatasmart=yes])
+
+AC_SUBST([WITH_LIBATASMART], [0])
+AM_CONDITIONAL(WITH_LIBATASMART, test "x$with_libatasmart" != "xno")
+AS_IF([test "x$with_libatasmart" != "xno"],
+      [AC_DEFINE([WITH_BD_SMART], [], [Define if smart is supported])
+       AC_SUBST([WITH_SMART], [1])
+       AC_SUBST([WITH_LIBATASMART], [1])],
+      [])
+
+AC_ARG_WITH([smartmontools],
+    AS_HELP_STRING([--with-smartmontools], [support ATA/SCSI S.M.A.R.T. via smartmontools @<:@default=no@:>@]),
+    [],
+    [with_smartmontools=no])
+
+AC_SUBST([WITH_SMARTMONTOOLS], [0])
+AM_CONDITIONAL(WITH_SMARTMONTOOLS, test "x$with_smartmontools" != "xno")
+AS_IF([test "x$with_smartmontools" != "xno"],
+      [AC_DEFINE([WITH_BD_SMART], [], [Define if smart is supported])
+       AC_SUBST([WITH_SMART], [1])
+       AC_SUBST([WITH_SMARTMONTOOLS], [1])],
+      [])
+
+AM_CONDITIONAL(WITH_SMART, test "x$with_libatasmart" != "xno" -o "x$with_smartmontools" != "xno" )
```

Comment:
I think having both `SMART` and `LIBATASMART` and `SMARTMONTOOLS` "plugins" (smart plugin and two implementations) would be confusing. We should pick one of the implementations and call it simply "smart" and the alternative "smartmontools" (if we decide that "libatasmart" should be default). If we drop libatasmart in the future we can simply rename the smartmontools implementation to just "smart".

Reply:
Done!